src/stages/headers.rs

```diff
@@ -547,32 +550,34 @@ impl HeaderDownload {
 
     async fn handle_response(
         node: Arc<Node>,
+        consensus: Arc<dyn Consensus>,
         requests: Arc<DashMap<BlockNumber, HeaderRequest>>,
         graph: Arc<Mutex<ForkChoiceGraph>>,
         peer_id: H512,
         response: BlockHeaders,
     ) {
         let cur_size = response.headers.len();
         debug!("Handling response from {peer_id} with {cur_size} headers");
-        let headers = Self::check_headers(&node, response.headers);
-        if cur_size != headers.len() {
-            node.penalize_peer(peer_id).await;
-        } else {
-            let key = headers[0].1.number;
-            let last_hash = headers[headers.len() - 1].0;
 
-            let mut graph = graph.lock();
+        match Self::check_headers(&consensus, response.headers) {
+            Ok(headers) => {
+                let key = headers[0].1.number;
```

Comment:
This will panic if `headers` empty?

Reply:
Yes, but we always check that `headers` are not empty, before passing them to this function.